test/augmentation/test_random_generator_3d.py

```diff
@@ -0,0 +1,561 @@
+import pytest
+import torch
+from torch.testing import assert_allclose
+
+from kornia.augmentation.random_generator import (
+    random_rotation_generator3d,
+    random_affine_generator3d,
+    random_motion_blur_generator3d,
+    center_crop_generator3d,
+    random_crop_generator3d,
+    random_perspective_generator3d,
+)
+
+
+class RandomGeneratorBaseTests():
+
+    def test_valid_param_combinations(self, device, dtype):
+        raise NotImplementedError
+
+    def test_invalid_param_combinations(self, device, dtype):
+        raise NotImplementedError
+
+    def test_random_gen(self, device, dtype):
+        raise NotImplementedError
+
+    def test_same_on_batch(self, device, dtype):
+        raise NotImplementedError
+
+
+class TestRandomPerspectiveGen3D(RandomGeneratorBaseTests):
+
+    @pytest.mark.parametrize('batch_size', [1, 8])
+    @pytest.mark.parametrize('depth,height,width', [(200, 200, 200)])
+    @pytest.mark.parametrize('distortion_scale', [torch.tensor(0.), torch.tensor(0.5), torch.tensor(1.)])
+    @pytest.mark.parametrize('same_on_batch', [True, False])
+    def test_valid_param_combinations(
+        self, depth, height, width, distortion_scale, batch_size, same_on_batch, device, dtype
+    ):
+        random_perspective_generator3d(
+            batch_size=batch_size, depth=depth, height=height, width=width,
+            distortion_scale=distortion_scale.to(device=device, dtype=dtype), same_on_batch=same_on_batch)
+
+    @pytest.mark.parametrize('depth,height,width,distortion_scale', [
+        # Should be failed if distortion_scale > 1. or distortion_scale < 0.
+        pytest.param(-100, 100, 100, torch.tensor(0.5), marks=pytest.mark.xfail),
```

Comment:
why `pytest.mark.fail` instead of `pytest.raises` ?
as in of https://github.com/kornia/kornia/blob/master/test/color/test_gray.py#L24

Reply:
```pytest.mark.xfail``` is recommended by the doc when working with parametrize: https://docs.pytest.org/en/stable/parametrize.html

Reply:
is this a potential fail or always fail ? I see you can pass to `pytest.raises` too.

Reply:
Those functions are always expected failing. We cannot pass ```pytest.raises``` to marks.